core/src/api/api.rs

```diff
@@ -7,19 +7,23 @@ use crate::{
 		dsnp_types::{DsnpGraphEdge, DsnpUserId},
 		encryption::EncryptionBehavior,
 	},
-	graph::{updates::UpdateEvent, user::UserGraph},
+	graph::{
+		key_manager::{PublicKeyManager, PublicKeyProvider, UserKeyProvider},
+		updates::UpdateEvent,
+		user::UserGraph,
+	},
 	iter_graph_connections,
 	util::time::time_in_ksecs,
 };
 use anyhow::{Error, Result};
-use std::{cmp::min, collections::HashMap, marker::PhantomData};
+use std::{cell::RefCell, cmp::min, collections::HashMap, marker::PhantomData, rc::Rc};
 
 const MAX_GRAPH_USERS_DEFAULT: usize = 1000;
 
-#[allow(non_snake_case)]
-#[derive(Debug, Clone, Eq, PartialEq)]
+#[derive(Debug)]
 pub struct GraphState<E: EncryptionBehavior, const MAX_USERS: usize = MAX_GRAPH_USERS_DEFAULT> {
```

Comment:
Based on your changes, I think we can completely get rid of the `<E: EncryptionBehavior>` generic parameter, pending outcome of discussion of the points you raised below.

Reply:
I think so, will do that in another PR that standardizes apis